docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,197 @@
+# Improving User Experience for Deploying and Upgrading Applications
+
+The goal of this document is to specify the design proposal for improving the current approach for deploying and upgrading applications.
+
+## Definition of the problem
+
+When users deploy an application we present the default `values.yaml` that the chart contains. This is the only mechanism they have to modify the default behavior. In many cases, they want to modify this default setup with simple modifications like changing the default hostname or modify the default password. It's not trivial to find these settings in the values.yaml that it's usually very verbose. They should be able to modify some "common" parameters easily. In any case, for advanced users, they should be able to still modify the `values.yaml` to fine-tune the settings of the application.
+
+Once an application is deployed, eventually the user will need to upgrade it. Now, apart from the complexity of having to handle the `values.yaml` file, it's necessary to remember the changes done in the original `values.yaml` in order to port those to the new version. In Kubeapps, once a user selects a new version, the values are substituted with the defaults for the new version, making it very difficult to modify these values with the previous one. It should be possible to follow the same approach than in the deployment: have an easy-to-fill form, having the possibility to switch to the advanced form in which they can modify the values as their liking.
+
+## Desired User Experience (UX)
+
+### Deployment and Upgrade Form
+
+When users reach the deployment form, the desired behavior is to have two different views, one for the basic form and another for an advanced deployment. That can be represented as two different tabs. The basic deployment should contain all the basic parameters while the advanced form will show the information as we do today. For the first version, changes made in the basic deployment form won't affect the advanced form and viceversa. Only when the user clicks on "submit" the current form information (either basic or advanced) will be sent to the backend.
+
+This is the proposed view for the basic deployment form:
+
+<img src="./img/deployment-form-basic.png" width="500px">
```

Comment:
What do you think about `Visual Editor` / `values.yaml` instead of `Basic` /  `Advanced`? To me, it sounds more close to the Helm terminology.

Reply:
`Visual Editor`? I am not sure what you mean with that. I am okay changing those names though.